extension/encoding/jsonlogencodingextension/extension.go

```diff
@@ -78,35 +132,23 @@ func (e *jsonLogExtension) Shutdown(_ context.Context) error {
 	return nil
 }
 
-func (e *jsonLogExtension) logProcessor(ld plog.Logs) ([]byte, error) {
-	logs := make([]logBody, 0, ld.LogRecordCount())
-
-	rls := ld.ResourceLogs()
-	for i := 0; i < rls.Len(); i++ {
-		rl := rls.At(i)
-		resourceAttrs := rl.Resource().Attributes().AsRaw()
-
-		sls := rl.ScopeLogs()
-		for j := 0; j < sls.Len(); j++ {
-			sl := sls.At(j)
-			logSlice := sl.LogRecords()
-			for k := 0; k < logSlice.Len(); k++ {
-				log := logSlice.At(k)
-				logEvent := logBody{
-					Body:               log.Body().AsRaw(),
-					ResourceAttributes: resourceAttrs,
-					LogAttributes:      log.Attributes().AsRaw(),
-				}
-				logs = append(logs, logEvent)
+// todDecodedJSONDocuments is a generic helper to derive json records decoded from a reader.
+func todDecodedJSONDocuments(reader io.Reader) ([]map[string]any, error) {
```

Comment:
It would be good to stream-decode, i.e. decode & process one object at a time. I think it's fine to come back to that though.

Reply:
Good suggestion and I see the value here : we reduce the memory usage of the map, which is sort of a temp buffer before appending to plog.

Give we are adding support for ndjson, and given such inputs can be large, I have addressed this through https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40932/commits/3f2f4d3d9adb31892a896379d41a41fc35c6c3cf :)